doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -66,10 +222,10 @@ The implementation requires introducing a new ``like=`` keyword to all existing
 array creation functions of NumPy. As examples of functions that would add this
 new argument (but not limited to) we can cite those taking array-like objects
 such as ``array`` and ``asarray``, functions that create arrays based on
-numerical ranges such as ``range`` and ``linspace``, as well as the ``empty``
-family of functions, even though that may be redundant, since there exists
-already specializations for those with the naming format ``empty_like``. As of
-the writing of this NEP, a complete list of array creation functions can be
+numerical inputs such as ``range`` and ``identity``, as well as the ``empty``
+family of functions, even though that may be redundant, since specializations
+for those already exist with the naming format ``empty_like``. As of the
```

Comment:
Question: will `empty_like(<dask array>)` be the same as `empty(<dask array>.shape, like=<dask array>)`?

If not, then for me that seems to be an argument in favour of changing `like=` to `typeof=` as discussed on the mailing list. (Apologies if this is the wrong forum for this and/or if I'm late to the party.)

If so, then that affects the backwards compatibility section rather dramatically?

Reply:
> Question: will `empty_like(<dask array>)` be the same as `empty(<dask array>.shape, like=<dask array>)`?

Yes, it's the same, which is why I wrote those are "redundant".

> If not, then for me that seems to be an argument in favour of changing `like=` to `typeof=` as discussed on the mailing list. (Apologies if this is the wrong forum for this and/or if I'm late to the party.)

Using `like=` was indeed due to the existing `empty_like` functions, `like` as an argument (proposed in NEP-35) has the same meaning as the `_like` suffix in a function.

> If so, then that affects the backwards compatibility section rather dramatically?

Given my answers above, it shouldn't.

Reply:
`empty_like` copies across the dtype, which is why I asked about that elsewhere in this PR..

Reply:
> empty_like copies across the dtype, which is why I asked about that elsewhere in this PR..

Yeah, that's a good point. It's still possible to get the same behavior, although it's not going to happen automatically as this is part of `empty_like` implementation itself.

Reply:
> If so, then that affects the backwards compatibility section rather dramatically?

Currently, calling np.empty_like on a dask array will give a numpy array, right? And you're saying this proposal will change that behaviour to return a dask array? That is a massive behaviour change!

Reply:
> Currently, calling np.empty_like on a dask array will give a numpy array, right? And you're saying this proposal will change that behaviour to return a dask array? That is a massive behaviour change!

No, currently calling `np.empty_like` on a Dask array will return a Dask array. This NEP will _not_ change `empty_like`'s behavior whatsoever. However, calling `np.empty` on a Dask array will return a NumPy array, and this NEP adds a `like=` argument to `np.empty` (but not to `np.empty_like`), which allows it to return a Dask array _iff_ `like=dask_array`.

Reply:
> No, currently calling np.empty_like on a Dask array will return a Dask array

🤯

Carry on. 😂